### PR TITLE
Issue 10 - Add Token Usage Option

### LINKED ISSUE #10: Add token info feature
Add a new command-line flag: `--token-usage` or `-t`. When the program is run with the `--token-usage`/`-t` flag set, extra information will be reported to `stderr` about the number of tokens that were sent in the prompt and returned in the completion.

### PR DESCRIPTION
**Pull request by mayank-Pareek**
## Description:
This pull request adds a new `--token-usage` flag (with a shorthand `-t` option) to the program, allowing users to see the token usage information when making requests to the API. This flag enables the program to display the number of prompt tokens, completion tokens, and total tokens used by the request.

### Changes Made:
- Added both a long flag `--token-usage` and a short flag `-t` to the program for reporting token usage.
- Added logic to check for the `--token-usage` flag in the program. If the flag is present and the response contains token usage data, the program will now extract and display it in the console using `console.error`.
- The relevant data is extracted from `chunk?.x_groq?.usage` in the response, following GROQ API's response structure.
- Updated the README.md file to document the new `--token-usage (-t)` option.

### Notes:
- No breaking changes were introduced in this pull request.
- This PR closes #10 by adding the requested token usage option.

Please let me know if any there are any additional changes required.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -23,33 +24,49 @@ program
...
+            }
           }
           await fs.writeFile(outputFile, `${response}\n`);
         } else {
-					// If no output file specified, read stream without storing to a variable
+          // If no output file specified, read stream without storing to a variable
           for await (const chunk of responseStream) {
             const chunkContent = chunk.choices[0]?.delta?.content || "";
             process.stdout.write(chunkContent);
+            //Record tokens if report flag passed
```
Comment (uday-rana) [inline src/index.js]: Add a space after `//` for readability

File: src/index.js [hunk 2]
```diff
@@ -23,33 +24,49 @@ program
...
           let response = "";
-					// Read response stream one chunk at a time
-					// Store chunks in `response` for writing to output file
+          // Read response stream one chunk at a time
+          // Store chunks in `response` for writing to output file
           for await (const chunk of responseStream) {
             const chunkContent = chunk.choices[0]?.delta?.content || "";
             process.stdout.write(chunkContent);
             response += chunkContent;
+            //Record tokens if report flag passed
```
Comment (uday-rana) [inline src/index.js]: Add a space after // for readability

File: src/index.js [hunk 3]
```diff
@@ -23,33 +24,49 @@ program
...
+              prompt_tokens = chunk.x_groq.usage.prompt_tokens;
+              completion_tokens = chunk.x_groq.usage.completion_tokens;
+              total_tokens = chunk.x_groq.usage.total_tokens;
+            }
           }
         }
+        if (reportToken) {
+          console.error(`\nPrompt tokens: ${prompt_tokens}`);
+          console.error(`Completion tokens: ${completion_tokens}`);
+          console.error(`Total tokens: ${total_tokens}`);
```
Comment (uday-rana) [inline src/index.js]: Combine into single `console.error` function call

**Review by uday-rana — CHANGES_REQUESTED** [review]
Works great overall, just need some minor changes in code style.

**Comment by mayank-Pareek** [comment]
Thanks for pointing out the required changes, I have committed some new changes to the branch, please review and suggest if this needs any more changes.

**Review by uday-rana — APPROVED** [review]
Looks good!

**Pull request merged**